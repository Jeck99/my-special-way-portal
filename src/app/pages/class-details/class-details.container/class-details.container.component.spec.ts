import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { ClassDetailsContainerComponent } from './class-details.container.component';
import { ClassDetailsViewComponent } from '../class-details.view/class-details.view.component';
import { NO_ERRORS_SCHEMA } from '@angular/core';

describe('ClassDetailsContainerComponent', () => {
  let component: ClassDetailsContainerComponent;
  let fixture: ComponentFixture<ClassDetailsContainerComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ClassDetailsContainerComponent],
      schemas: [NO_ERRORS_SCHEMA],
    })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ClassDetailsContainerComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should create empty matrix of lessons given days and hours counts and empty timeslots array', () => {
    component = new ClassDetailsContainerComponent();
    const schedule = component.buildScheduleFromTimeslots(2, 2, []);

    expect(schedule.length).toBe(2);
    expect(schedule[0].length).toBe(2);
  });

  it('should create matrix with lesson in 0,0 place given days and hours and timeslot array[1]', () => {
    component = new ClassDetailsContainerComponent();
    const schedule = component.buildScheduleFromTimeslots(2, 2, [{
      index: '10',
      lesson: {
        _id: 'someId',
        icon: 'some icon',
        title: 'some title',
      },
    }]);

    expect(schedule[1][0]).toEqual({
      _id: 'someId',
      icon: 'some icon',
      title: 'some title',
    });
    expect(schedule.length).toBe(2);
    expect(schedule[0].length).toBe(2);
  });
});
