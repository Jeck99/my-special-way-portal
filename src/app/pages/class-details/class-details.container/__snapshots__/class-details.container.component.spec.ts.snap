// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassDetailsContainerComponent should create 1`] = `
ClassDetailsContainerComponent {
  "_class": Object {
    "_id": "5b217b030825622c97d3757f",
    "level": "א",
    "name": "טיטאן",
    "number": 1,
    "schedule": Array [
      Object {
        "index": "10",
        "lesson": Object {
          "_id": "someid",
          "icon": "someicon",
          "title": "soetitle",
        },
      },
    ],
  },
  "classService": Object {
    "classById": [MockFunction] {
      "calls": Array [
        Array [
          "5b217b030825622c97d3757f",
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": ZoneAwarePromise {
            "__zone_symbol__state": true,
            "__zone_symbol__value": Object {
              "data": Object {
                "classById": Object {
                  "_id": "5b217b030825622c97d3757f",
                  "level": "א",
                  "name": "טיטאן",
                  "number": 1,
                  "schedule": Array [
                    Object {
                      "index": "10",
                      "lesson": Object {
                        "_id": "someid",
                        "icon": "someicon",
                        "title": "soetitle",
                      },
                    },
                  ],
                },
              },
            },
          },
        },
      ],
    },
    "update": [MockFunction],
  },
  "dialog": Object {
    "open": [MockFunction],
  },
  "route": Object {
    "snapshot": Object {
      "params": Object {
        "id": "5b217b030825622c97d3757f",
      },
    },
  },
  "schedule": Array [
    Array [
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ],
    Array [
      Object {
        "_id": "someid",
        "icon": "someicon",
        "title": "soetitle",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ],
    Array [
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ],
  ],
  "scheduleService": Object {
    "daysLabels": Array [
      "ראשון",
      "שני",
      "שלישי",
      "רביעי",
      "חמישי",
      "שישי",
    ],
    "hoursLabels": Array [
      "07:30 - 08:00",
      "08:00 - 08:50",
      "08:50 - 09:35",
    ],
    "levels": Array [
      "א",
      "ב",
      "ג",
    ],
  },
}
`;
